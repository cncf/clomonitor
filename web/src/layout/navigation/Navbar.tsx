import { Dispatch, SetStateAction } from 'react';
import { Link } from 'react-router-dom';

import logo from '../../media/clomonitor.svg';
import ThemeSwitch from '../common/ThemeSwitch';
import Searchbar from '../navigation/Searchbar';
import styles from './Navbar.module.css';

interface Props {
  setScrollPosition: Dispatch<SetStateAction<number | undefined>>;
}

const Navbar = (props: Props) => {
  return (
    <nav className={`navbar ${styles.navbar}`}>
      <div className="container-lg">
        <div className="d-flex flex-column flex-md-row align-items-center justify-content-between w-100">
          <div className="me-0 me-md-4 mt-2 mt-md-0">
            <div className="d-flex flex-row align-items-start">
              <Link to="/" onClick={() => props.setScrollPosition(0)} className="cursorPointer">
                <img className={styles.logo} alt="CLOMonitor logo" src={logo} />
              </Link>
              <div className={`ms-1 badge rounded-0 text-uppercase ${styles.badge} alphaBadge`}>Alpha</div>
            </div>
          </div>
          <div className={`d-flex flex-row align-items-center ${styles.searchWrapper}`}>
            <Searchbar classNameWrapper="my-3 flex-grow-1" setScrollPosition={props.setScrollPosition} />
            <ThemeSwitch />
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
