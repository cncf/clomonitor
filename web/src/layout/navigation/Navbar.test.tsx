import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';

import Navbar from './Navbar';

const mockSetScrollPosition = jest.fn();

const defaultProps = {
  setScrollPosition: mockSetScrollPosition,
};

describe('Navbar', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(
      <Router>
        <Navbar {...defaultProps} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(
      <Router>
        <Navbar {...defaultProps} />
      </Router>
    );

    expect(screen.getByText('Alpha')).toBeInTheDocument();
    expect(screen.getByAltText('CLOMonitor logo')).toBeInTheDocument();

    const link = screen.getByRole('link');
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/');

    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByRole('switch')).toBeInTheDocument();
  });

  it('clicks logo', () => {
    render(
      <Router>
        <Navbar {...defaultProps} />
      </Router>
    );

    const link = screen.getByRole('link');
    userEvent.click(link);

    expect(mockSetScrollPosition).toHaveBeenCalledTimes(1);
    expect(mockSetScrollPosition).toHaveBeenCalledWith(0);

    expect(window.location.pathname).toBe('/');
  });
});
