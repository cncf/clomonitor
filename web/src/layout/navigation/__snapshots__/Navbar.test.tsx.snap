// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar creates snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar"
  >
    <div
      class="container-lg"
    >
      <div
        class="d-flex flex-column flex-md-row align-items-center justify-content-between w-100"
      >
        <div
          class="me-0 me-md-4 mt-2 mt-md-0"
        >
          <div
            class="d-flex flex-row align-items-start"
          >
            <a
              class="cursorPointer"
              href="/"
            >
              <img
                alt="CLOMonitor logo"
                class="logo"
                src=""
              />
            </a>
            <div
              class="ms-1 badge rounded-0 text-uppercase badge alphaBadge"
            >
              Alpha
            </div>
          </div>
        </div>
        <div
          class="d-flex flex-row align-items-center searchWrapper"
        >
          <div
            class="position-relative my-3 flex-grow-1"
          >
            <div
              class="d-flex align-items-center overflow-hidden searchBar lh-base bg-white mx-auto searchBar search"
            >
              <input
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="flex-grow-1 ps-2 ps-md-3 border-0 shadow-none bg-transparent lh-base input"
                data-testid="search-bar"
                placeholder="Search projects"
                spellcheck="false"
                type="text"
                value=""
              />
              <button
                aria-label="Search text"
                class="btn btn-link lh-1 px-2 btnIcon"
              >
                <div
                  class="iconWrapper lightIcon"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
          <div
            class="form-check form-switch ms-4 position-relative d-none d-sm-block cursorPointer"
          >
            <div>
              <svg
                class="text-light position-absolute sunIcon"
                data-testid="sun-icon"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12 16C14.2091 16 16 14.2091 16 12C16 9.79086 14.2091 8 12 8C9.79086 8 8 9.79086 8 12C8 14.2091 9.79086 16 12 16ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M11 0H13V4.06189C12.6724 4.02104 12.3387 4 12 4C11.6613 4 11.3276 4.02104 11 4.06189V0ZM7.0943 5.68018L4.22173 2.80761L2.80752 4.22183L5.6801 7.09441C6.09071 6.56618 6.56608 6.0908 7.0943 5.68018ZM4.06189 11H0V13H4.06189C4.02104 12.6724 4 12.3387 4 12C4 11.6613 4.02104 11.3276 4.06189 11ZM5.6801 16.9056L2.80751 19.7782L4.22173 21.1924L7.0943 18.3198C6.56608 17.9092 6.09071 17.4338 5.6801 16.9056ZM11 19.9381V24H13V19.9381C12.6724 19.979 12.3387 20 12 20C11.6613 20 11.3276 19.979 11 19.9381ZM16.9056 18.3199L19.7781 21.1924L21.1923 19.7782L18.3198 16.9057C17.9092 17.4339 17.4338 17.9093 16.9056 18.3199ZM19.9381 13H24V11H19.9381C19.979 11.3276 20 11.6613 20 12C20 12.3387 19.979 12.6724 19.9381 13ZM18.3198 7.0943L21.1923 4.22183L19.7781 2.80762L16.9056 5.6801C17.4338 6.09071 17.9092 6.56608 18.3198 7.0943Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <input
              checked=""
              class="form-check-input cursorPointer themeSwitch"
              id="theme"
              role="switch"
              type="checkbox"
            />
          </div>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
